lighthouse-core/report/handlebar-helpers.js

```diff
@@ -0,0 +1,233 @@
+    return Math.round(aggregation.total * 100);
+  }
+
+  // Converts a value to a rating string, which can be used inside the report
+  // for color styling.
+  static getItemRating(value) {
+    if (typeof value === 'boolean') {
+      return value ? RATINGS.GOOD.label : RATINGS.POOR.label;
+    }
+    return HandlebarHelpers.calculateRating(value);
+  }
+
+  // Converts a name to a link.
+  static nameToLink(name) {
+    return name.toLowerCase().replace(/\s/g, '-');
+  }
+
+  // Converts the total score to a rating that can be used for styling.
+  static totalScoreRating(aggregation) {
+    const totalScore = HandlebarHelpers.getTotalScore(aggregation);
+    return HandlebarHelpers.calculateRating(totalScore);
+  }
+
+  static shouldShowHelpText(value) {
+    return HandlebarHelpers.getItemRating(value) !== RATINGS.GOOD.label;
+  }
+
+
+  // Convert numbers to fixed point decimals
+  static decimal(number) {
+    if (number && number.toFixed) {
+      return number.toFixed(2);
+    }
+    return number;
+  }
+
+  // value is boolean?
+  static isBool(value) {
+    return typeof value === 'boolean';
+  }
+
+  // !value
+  static not(value) {
+    return !value;
+  }
+
+  // value === value2?
+  static ifEq(lhs, rhs, options) {
+    if (lhs === rhs) {
+      // eslint-disable-next-line no-invalid-this
+      return options.fn(this);
+    } else {
+      // eslint-disable-next-line no-invalid-this
+      return options.inverse(this);
+    }
+  }
+
+  // value !== value2
+  static ifNotEq(lhs, rhs, options) {
+    if (lhs !== rhs) {
+      // eslint-disable-next-line no-invalid-this
+      return options.fn(this);
+    } else {
+      // eslint-disable-next-line no-invalid-this
+      return options.inverse(this);
+    }
+  }
+
+  // arg1 && arg2 && ... && argn
+  static and(...args) {
+    let arg = false;
+    for (let i = 0, n = args.length - 1; i < n; i++) {
+      arg = args[i];
+      if (!arg) {
+        break;
+      }
+    }
+    return arg;
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  static sanitize(str, opts) {
+    // const isViewer = opts.data.root.reportContext === 'viewer';
+
+    // Allow the report to inject HTML, but sanitize it first.
+    // Viewer in particular, allows user's to upload JSON. To mitigate against
+    // XSS, define a renderer that only transforms links and code snippets.
+    // All other markdown ad HTML is ignored.
+    const renderer = new marked.Renderer();
+    renderer.em = str => `<em>${str}</em>`;
+    renderer.link = (href, title, text) => {
+      title = title || text;
+      return `<a href="${href}" target="_blank" rel="noopener" title="${title}">${text}</a>`;
+    };
+    renderer.codespan = function(str) {
+      return `<code>${str}</code>`;
+    };
+    // eslint-disable-next-line no-unused-vars
+    renderer.code = function(code, language) {
+      return `<pre>${code}</pre>`;
+    };
+    // Nuke wrapper <p> tag that gets generated.
+    renderer.paragraph = function(str) {
```

Comment:
This is missing `renderer.image`: https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/report/report-generator.js#L147.

Are you refactoring off the latest master? That one recently landed.